packages/next/src/utils.ts

```diff
@@ -1254,21 +1254,38 @@ async function getSourceFilePathFromPage({
     const extensionless = fsPath.replace(path.extname(fsPath), '');
 
     for (const ext of extensionsToTry) {
-      fsPath = `${extensionless}.${ext}`;
+      // for appDir, we need to treat "index.js" as root-level "page.js"
+      fsPath =
+        extensionless === path.join(workPath, 'app/index') ||
+        extensionless === path.join(workPath, 'src/app/index')
```

Comment:
Is the check for `src` here different than line 1247?

Reply:
It is for a slightly different purpose. Though, actually, we might need to update `usesSrcDirectory` to also check `src/app` path. Let me include that in this PR.

Reply:
@styfle made some tiny optimizations 👍